### PR TITLE
End-to-end test for configuring the Ethereum oracle

### PR DESCRIPTION
**Pull request by james-chf**
Closes https://github.com/anoma/namada/issues/686

This PR adds an end-to-end test checking that the Ethereum event oracle uses the configured values from blockchain storage (implementation already merged in https://github.com/anoma/namada/pull/810). To do this, an oracle-related `Config` struct had to be moved from `namada_apps` to `namada_ethereum_bridge`, to be accessible in `namada_tests`.

### CODE REVIEW

File: core/src/lib.rs [hunk 1]
```diff
@@ -12,7 +12,7 @@ pub mod ledger;
 pub mod proto;
 pub mod types;
 
-#[cfg(feature = "abciplus")]
+#[cfg(not(feature = "abcipp"))]
```
Comment (james-chf) [inline core/src/lib.rs:15]: Had to change this otherwise `make check-abcipp` stopped working

File: tests/src/e2e/eth_bridge_tests.rs [hunk 2]
```diff
@@ -237,3 +240,69 @@ fn test_add_to_bridge_pool() {
...
+            genesis
+        },
+        None,
+    )?;
+
+    // start the ledger with the real oracle and wait for a block to be
+    // committed
+
+    // TODO(namada#1061): need to start up a fake Ethereum node here for the
+    // oracle to connect to, to avoid errors in the ledger logs
```
Comment (james-chf) [inline tests/src/e2e/eth_bridge_tests.rs:281]: At the moment, the `namadan` logs gets spammed with error messages about the oracle not being able to connect to an Ethereum RPC endpoint, for this test it doesn't matter as we are just testing the oracle is initially configured correctly, but when the linked issue [namada#1061](https://github.com/anoma/namada/issues/1061) is done we can change this here also)

File: core/src/lib.rs (hunk 1 above)
Comment (sug0) [reply core/src/lib.rs:15]: for the sake of consistency, I would even rename every feature flag to `#[cfg(not(feature = "abcipp"))]`. obviously that's out of the scope of this PR, though

**Review by sug0 — APPROVED** [review]
lgtm

File: apps/Cargo.toml [hunk 3]
```diff
@@ -74,6 +74,7 @@ abciplus = [
 
 [dependencies]
 namada = {path = "../shared", default-features = false, features = ["wasm-runtime", "ferveo-tpke"]}
+namada_ethereum_bridge = {path = "../ethereum_bridge"}
```
Comment (batconjurer) [inline apps/Cargo.toml:77]: The pattern here is to re-export needed deps inside of the shared sub-crate. You'll notice that PoS does the same.

**Review by batconjurer — CHANGES_REQUESTED** [review]

File: apps/Cargo.toml (hunk 3 above)
Comment (james-chf) [reply apps/Cargo.toml]: oic, have updated to use `namada::eth_bridge` instead

Comment (james-chf) [reply apps/Cargo.toml]: updated in `tests` crate also now

**Review by batconjurer — APPROVED** [review]

**Pull request merged**